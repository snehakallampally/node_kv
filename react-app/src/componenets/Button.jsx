import React from "react";

const Button = (props) => {

  return (
    <button className={props.className} type={props.type} data-testid="button-test-id" onClick={props.onClick}>
      {props.label}
    </button>
  );
};
export default Button;
