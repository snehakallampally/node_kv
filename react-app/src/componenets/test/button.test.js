import React from "react";
import { getByTestId, getByText, render,fireEvent } from "@testing-library/react";
import Button from "../Button";

describe("Check if button is working",()=>{
test("First", () => {
    const {getByTestId}=render(<Button/>)

    const element=getByTestId("button-test-id");
    console.log

    expect(element).toBeTruthy()
});

test("check if button text content is rendering properly", () => {
const text="Click";

    const {getByText}=render (<Button label={text}/>)

    getByText(text);
});
test("check if nbutton click is working",()=>{
    const onClick=jest.fn();

    const {getByTestId}=render(<Button onClick={onClick}/>);
    const element=getByTestId("button-test-id")

    fireEvent.click(element);

    expect(onClick).toHaveBeenCalled()

});

test("check if snapshots are matching properly",()=>{
    const onClick=jest.fn();
    const text="Click";

    const {asFragment}=render(<Button label={text} onClick={onClick}/>)
    
    expect(asFragment()).toMatchSnapshot();
})
})