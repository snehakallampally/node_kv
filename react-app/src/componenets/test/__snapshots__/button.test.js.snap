// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if button is working check if snapshots are matching properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-test-id"
  >
    Click
  </button>
</DocumentFragment>
`;
